### PR TITLE
/getitem - Also get items by dbName

### PR DESCRIPTION
**Pull request by ItsSpencer**
Related Issue: https://github.com/Stuyk/altv-athena/issues/296

### CODE REVIEW

File: src/core/plugins/core-commands/server/commands/moderator/inventory.ts [hunk 1]
```diff
@@ -41,19 +41,26 @@ class InventoryCommands {
 
     @command('getitem', LocaleController.get(LOCALE_KEYS.COMMAND_GET_ITEM, '/getitem'), PERMISSIONS.ADMIN)
     private static async handleGetItem(player: alt.Player, ...args) {
+        const amount = args[args.length - 1];
```
Comment (Stuyk) [inline src/core/plugins/core-commands/server/commands/moderator/inventory.ts:44]: args can potentially have no length resulting in an error here

File: src/core/plugins/core-commands/server/commands/moderator/inventory.ts [hunk 2]
```diff
@@ -41,19 +41,26 @@ class InventoryCommands {
...
+
+        args.pop();
```
Comment (Stuyk) [inline src/core/plugins/core-commands/server/commands/moderator/inventory.ts:46]: should only pop if the last argument is a number and should verify length first
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN

if not present, default the amount to 1

**Review by Stuyk — CHANGES_REQUESTED** [review]

**Comment by ItsSpencer** [comment]
I have implemented your change requests so far :)

File: src/core/plugins/core-commands/server/commands/moderator/inventory.ts [hunk 3]
```diff
@@ -41,19 +41,38 @@ class InventoryCommands {
...
-        const fullItemName = args.join(' ');
+        if (args.length === 0) {
+            return Athena.player.emit.message(player, LocaleController.get(LOCALE_KEYS.ITEM_ARGUMENTS_MISSING));
+        }
+
+        if (args.length === 1) {
+            var amount: number = 1;
+            var fullItemName: string = args[0];
+        } else if (args.length >= 2 && isNaN(args[args.length - 1])) {
+            var amount: number = 1;
```
Comment (Booster1212) [inline src/core/plugins/core-commands/server/commands/moderator/inventory.ts:52]: var means you're in global scope somehow, so i'd rather replace it by let / const, depending if you want to change it or not

File: src/core/plugins/core-commands/server/commands/moderator/inventory.ts [hunk 4]
```diff
@@ -41,19 +41,38 @@ class InventoryCommands {
 
     @command('getitem', LocaleController.get(LOCALE_KEYS.COMMAND_GET_ITEM, '/getitem'), PERMISSIONS.ADMIN)
     private static async handleGetItem(player: alt.Player, ...args) {
-        const fullItemName = args.join(' ');
+        if (args.length === 0) {
+            return Athena.player.emit.message(player, LocaleController.get(LOCALE_KEYS.ITEM_ARGUMENTS_MISSING));
+        }
+
+        if (args.length === 1) {
```
Comment (Stuyk) [inline src/core/plugins/core-commands/server/commands/moderator/inventory.ts:48]: I think this if statement can have way reduced complexity.

Here's a TypeScript example.
https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgJwOYRgXhgbQOQAsCmANoSNgDQzYDuIChAJuZVPtE9gKzYC6A3ALAAoIYXyw4AWxABXMLAwBGAYICWAMxgAKRCgB0osEii4YAPgwAmAJQwA3kJiOYoSLAAOIFnJVxCAQSlZeRg3RAh8AEk5bWQITB0IfXxDYxgAWhgFbitlJxh1LQBCFQgAOThSzQ8vKB9-QLkrG3tBPLzJGTl0EM9k2t8AzqhctscE3Q83TRyHJwBfIQXhQRdofJYJcol8bvGAKxAVME1sSisgA

**Review by Stuyk — CHANGES_REQUESTED** [review]

**Comment by ItsSpencer** [comment]
refactored that part

**Review by Stuyk — APPROVED** [review]

**Pull request merged**